### PR TITLE
Add support for very long command lines in ntosebpfext process data

### PR DESCRIPTION
**Pull request by Austin-Lamb**
## Description
Previously we were limited to a few hundred characters for command lines.  This isn't long enough in many cases - such as `cl.exe` compiler invocations that have long include paths on them.  This uses the new support in 0.17 to help with marshaling long strings, so I allow up to 32k characters now which I believe is the maximum possible on Windows because `UNICODE_STRING` has a `USHORT MaximumLength` and that is the byte count, so character count is "max USHORT / 2" or 32k.

## Testing
Added a test that verifies a process with very long arguments (>5k characters) comes through as expected.

## Documentation
No docs need updated AFAIK.

## Installation
No installer impact.

### CODE REVIEW

File: tools/process_monitor_bpf/process_monitor.c [hunk 1]
```diff
@@ -8,6 +8,26 @@
 #include "bpf_helpers.h"
 #include "ebpf_ntos_hooks.h"
 
+#define IMAGE_PATH_SIZE (1024)
+#define COMMAND_SCRATCH_SIZE (32 * 1024) // 32k characters is the max char count that fits in a UNICODE_STRING
```
Comment (Alan-Jowett) [inline tools/process_monitor_bpf/process_monitor.c:12]: Did we switch this provider over to using UTF16? If we did, do we need to increase this to account for 32K uint16?

Comment (Austin-Lamb) [reply tools/process_monitor_bpf/process_monitor.c:12]: It's still on UTF-8 for now, and I've been torn about how to feel about that.  If it was UTF-16 that'd be nice for my use cases which often intersect with .NET where strings are natively UTF-16.  But more and more of Windows is embracing UTF-8 so I think that could also remain the right choice.

Do you have opinions on which UTF-* we should be using?  I could swap to UTF-16 if preferred, but I'd like to do that in a subsequent PR.

Comment (Alan-Jowett) [reply tools/process_monitor_bpf/process_monitor.c:12]: The extension shouldn't be futzing with the string IMHO. It should just pass it through. The only reason I did it was to pack mor characters into the 512 byte stack limit.

Comment (Austin-Lamb) [reply tools/process_monitor_bpf/process_monitor.c:12]: ok, I'll move to UTF-16 in a followup PR.

**Pull request merged**